vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/Configuration.java

```diff
@@ -39,6 +42,8 @@ public String getType() {
      *
      * @return the list of layers managed by this map
      */
+    @JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
+    @JsonIdentityReference(alwaysAsId = true)
```

Comment:
All nested objects are now just serialized as IDs. Since we sync every object separately and then just wire up the references, we don't need the full object anymore. In fact it would be wasteful to use the full object, as we would include some objects several times in the JSON that is sent to the client.